agent/lib/kontena/workers/container_log_worker.rb

```diff
@@ -67,7 +68,7 @@ def on_message(id, stream, chunk)
         type: stream,
         data: data
       }
-      @queue << msg
+      @queue << msg if @queue.size < QUEUE_LIMIT
```

Comment:
As a user I'd really like to see this action taking place in logs. Maybe a warning?